src/rqt_plot/rosplot.py

```diff
@@ -158,9 +158,10 @@ def _get_data(self, msg):
                 if isinstance(val, Bool):
                     # extract boolean field from bool messages
                     val = val.data
-                return float(val)
-            for f in self.field_evals:
-                val = f(val)
+            else:
+                for f in self.field_evals:
+                    val = f(val)
+            self.error = None
```

Comment:
Maybe the error should be reset in the `_ros_cb` method after acquiring the lock instead?

Reply:
@dirk-thomas Thank you for the comment! Updated the commit.